### PR TITLE
Add passing 3.11 CI by exempting blackd tests

### PR DESCRIPTION
**Pull request by cooperlees**
- Had to exempt blackd tests for now due to aiohttp
  - Skip by using `sys.version_info` tuple
  - aiohttp does not compile in 3.11 yet - refer to #3230
- Add a deadsnakes ubuntu workflow to run 3.11-dev to ensure we don't regress
  - Have it also format ourselves

Test:
- `tox -e 311`

### Checklist - did you ...

- [x] Add a CHANGELOG entry if necessary?
- [x] Add / update tests if necessary?
- [x] Add new / update outdated documentation?

### CODE REVIEW

File: .github/workflows/test-311.yml [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+        with:
+          python-version: ${{ matrix.python-version }}
+      - uses: deadsnakes/action@v2.1.1
+        if: endsWith(matrix.python-version, '-dev')
+        with:
+          python-version: ${{ matrix.python-version }}
+          # debug: true  # Optional, to select a Python debug build
+
+      - name: Show python version
+        run: python --version --version && which python
```
Comment (JelleZijlstra) [inline .github/workflows/test-311.yml:37]: huh, I didn't know you could repeat `--version` and get more detail

File: CHANGES.md [hunk 2]
```diff
@@ -69,6 +69,8 @@
 
 <!-- Changes to how Black is packaged, such as dependency requirements -->
 
+- Add 3.11 CI + declare 3.11 supported minus blackd (#3234)
```
Comment (JelleZijlstra) [inline CHANGES.md:72]:
```suggestion
- Python 3.11 is now supported, except for `blackd` (#3234)
```

File: tests/test_blackd.py [hunk 3]
```diff
@@ -7,195 +8,199 @@
 
 from tests.util import DETERMINISTIC_HEADER, read_data
```
Comment (JelleZijlstra) [inline tests/test_blackd.py:10]: I think you can use https://docs.pytest.org/en/7.1.x/how-to/skipping.html#skip-all-test-functions-of-a-class-or-module so you don't have to indent the whole thing

File: .github/workflows/test-311.yml [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+
+permissions:
+  contents: read
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        python-version: ["3.11-dev"]
```
Comment (skshetry) [inline .github/workflows/test-311.yml:22]: `3.11.0-rc` is available in `actions/setup-python@v4`. Unless I am missing something, you don't have to use `deadsnakes/action`.

```suggestion
        python-version: ["3.11.0-rc - 3.11"]
```

This should provide any latest officially released version of Python 3.11.

**Review by ichard26 — APPROVED** [review]
Nothing blocking. Just quality of life suggestions really.

Thank you for the PR, I really appreciate it! You are a good maintainer of Black, it's just that your attention is divided elsewhere, and that's OK.

**Comment by cooperlees** [comment]
Sigh, ci-311 broke it ...

**Comment by ichard26** [comment]
Just revert, it doesn't matter anymore. It's not worth spending a second more improving this PR. I'm sorry, I have wasted way too much of your time.

**Comment by ichard26** [comment]
Feel free to merge. Thanks again for your time!

**Pull request merged**